library/cartridge_edit_topology.py

```diff
@@ -407,51 +454,96 @@ def get_servers_params(instances, cluster_instances, allow_missed_instances):
 
-    servers_params, err = get_servers_params(instances, cluster_instances, allow_missed_instances)
-    if err is not None:
-        return None, err
+    if extra_cluster_instances and not ignore_checks.get(EXTRA_CLUSTER_INSTANCES_CHECK_NAME):
+        msg = 'some instances from cluster are missing in inventory: %s' % ', '.join(extra_cluster_instances)
+        errors.append(msg)
 
-    if servers_params:
-        topology_params['servers'] = servers_params
+    if changed_advertise_uris and not ignore_checks.get(ADVERTISE_URIS_CHANGE_CHECK_NAME):
+        msg = 'advertise uris of some instances changed: %s' % ', '.join(changed_advertise_uris)
+        errors.append(msg)
 
-    return topology_params, None
+    return errors
 
 
-def get_replicasets_failover_priority_and_instances_params(
-    replicasets, cluster_replicasets, instances, cluster_instances, allow_missed_instances
-):
-    topology_params = {}
+def check_new_replicasets_for_dangerous_changes(all_new_replicasets, new_replicasets, old_replicasets, ignore_checks):
+    errors = []
+    extra_cluster_replicasets = []
+    renamed_replicasets = []
 
-    replicasets_params, err = get_replicasets_params_for_changing_failover_priority(
-        replicasets, cluster_replicasets, cluster_instances, allow_missed_instances
-    )
-    if err is not None:
-        return None, err
+    for old_replicaset_name, old_replicaset in old_replicasets.items():
+        if old_replicaset_name not in all_new_replicasets:
+            renamed = False
 
-    if replicasets_params:
-        topology_params['replicasets'] = replicasets_params
+            for new_replicaset_name, new_replicaset in all_new_replicasets.items():
+                if set(new_replicaset.get('instances', [])) == set(old_replicaset.get('instances', [])):
+                    renamed = True
 
-    # manage instances that were joined on previous call
-    servers_params, err = get_servers_params(instances, cluster_instances, allow_missed_instances)
-    if err is not None:
-        return None, err
+                    # Ignore renamed replicaset if it isn't in play hosts
+                    if new_replicaset_name in new_replicasets:
+                        renamed_replicasets.append('%s -> %s' % (new_replicaset_name, old_replicaset_name))
 
-    if servers_params:
-        topology_params['servers'] = servers_params
+            if not renamed:
+                extra_cluster_replicasets.append(old_replicaset_name)
 
-    return topology_params, None
+    if extra_cluster_replicasets and not ignore_checks.get(EXTRA_CLUSTER_REPLICASETS_CHECK_NAME):
+        msg = 'some replicasets from cluster are missing in inventory: %s' % ', '.join(extra_cluster_replicasets)
+        errors.append(msg)
+
+    if renamed_replicasets and not ignore_checks.get(RENAMED_REPLICASETS_CHECK_NAME):
+        msg = 'looks like that some replicasets should be renamed in inventory: %s' % ', '.join(renamed_replicasets)
```

Comment:
`should be renamed` -> `has different names`?

Reply:
No, it's a hint for user